src/node/blockstorage.h

```diff
@@ -216,6 +216,11 @@ class BlockManager
     //! Returns last CBlockIndex* that is a checkpoint
     const CBlockIndex* GetLastCheckpoint(const CCheckpointData& data) EXCLUSIVE_LOCKS_REQUIRED(cs_main);
 
+    //! Returns 'lower_block' if all blocks up to that point are available in disk, otherwise returns the first block that is not pruned in the upper-lower range.
+    //! If 'lower_block=nullptr': verify blocks availability up to the genesis block.
+    //! Returns nullptr if 'lower_block' is not an 'upper_block' ancestry.
+    const CBlockIndex* CheckBlockDataAvailability(const CBlockIndex& upper_block LIFETIMEBOUND, const CBlockIndex* lower_block=nullptr) EXCLUSIVE_LOCKS_REQUIRED(::cs_main);
```

Comment:
In commit "refactor: simplify pruning violation check" (2163b736f255dd5381a6f9ba4ad7dd32192c74e6)

This commit seems like a nice code simplification to the `BaseIndex::Init` function. I wonder if simplifying Init is the real motivation for the change at this point, or if this is also used for something else later. Would be helpful if commit message had a sentence about the benefit / motivation.

Also commit message should be updated to mention `CheckBlockDataAvailability` instead of `IsBlockDataAvailable`.

Also, it would be good to have a simple unit test for the `CheckBlockDataAvailability` function in `blockmanager_tests.cpp`. Even if the unit test doesn't cover all the edge cases, having it would be useful to show  `CheckBlockDataAvailability` makes sense as an API and is testable

EDIT: Now that I've reached the last commit of the PR  "index: verify blocks data existence only once" (ce5f798b089dc8051f6b414ce1a606e60c85d7f3) I can see that the motivation for commit is to be able to help make index startup more asynchronous, avoid repeating work when multiple indexes are enabled in the followup commit. Could mention something like that in the commit message here.